### PR TITLE
IT-5123 tab item

### PR DESCRIPTION
**Pull request by pioziol123**

### CODE REVIEW

File: lib/js/components/TabItem/TabItem.vue [hunk 1]
```diff
@@ -49,18 +94,24 @@ export default {
...
 		},
+		size: {
+			type: String,
+			default: TAB_ITEM_SIZES.MEDIUM,
+		},
 	},
 	data() {
 		return {
-			ICON_SIZES: Object.freeze(ICON_SIZES),
+			iconSize:
```
Comment (kvas-damian) [inline lib/js/components/TabItem/TabItem.vue:112]: this should be computed. What if I change the size on a instance that is already mounted?

File: lib/js/components/TabItem/TabItem.vue [hunk 2]
```diff
@@ -1,40 +1,85 @@
...
+		class="tabItem"
+		:title="label"
+		:class="{
+			'-sizeMedium': size === TAB_ITEM_SIZES.MEDIUM,
+			'-sizeSmall': size === TAB_ITEM_SIZES.SMALL,
+			'-isSelected': isSelected,
+		}"
+		@click="$emit('click')"
+	>
+		<wnl-icon class="tabItem__icon" :icon="icon" :size="iconSize" />
```
Comment (kvas-damian) [inline lib/js/components/TabItem/TabItem.vue:12]: we don't use `wnl-` prefix anymore

File: lib/js/components/TabItem/TabItem.vue [hunk 3]
```diff
@@ -1,40 +1,85 @@
...
 	&:hover {
 		border-bottom-color: $color-default-border;
-		color: $color-default-icon;
+
+		#{$self}__icon {
+			color: $color-default-icon;
+		}
+
+		#{$self}__label {
+			color: $color-default-text;
```
Comment (kvas-damian) [inline lib/js/components/TabItem/TabItem.vue:47]: maybe we could add a transition on these color change?

File: lib/js/components/TabItem/TabItem.stories.ts [hunk 4]
```diff
@@ -13,26 +14,39 @@ const StoryTemplate: StoryFn<typeof TabItem> = (args) => ({
...
 
 const args = {
-	isActive: false,
-	title: 'Lorem ipsum dolor sit amet',
+	isSelected: false,
 } as Args;
 
 const argTypes = {
 	icon: {
 		control: { type: 'select', options: Object.keys(ICONS) },
```
Comment (kvas-damian) [inline lib/js/components/TabItem/TabItem.stories.ts:35]: Please add `null` so we can deselect icon in this story

File: lib/js/components/TabItem/TabItem.vue (hunk 3 above)
Comment (pioziol123) [reply lib/js/components/TabItem/TabItem.vue:47]: something like `transition: color ease-in-out $default-transition-time;`?

Comment (kvas-damian) [reply lib/js/components/TabItem/TabItem.vue:47]: yes, similar to `border-bottom-color`

File: lib/js/components/TabItem/TabItem.vue [hunk 5]
```diff
@@ -1,67 +1,122 @@
...
+		:title="label"
+		:class="{
+			'-sizeMedium': size === TAB_ITEM_SIZES.MEDIUM,
+			'-sizeSmall': size === TAB_ITEM_SIZES.SMALL,
+			'-isSelected': isSelected,
+		}"
+		@click="$emit('click')"
+	>
+		<ds-icon v-if="icon !== null" class="tabItem__icon" :icon="icon" :size="iconSize" />
+		<span class="tabItem__label">{{ label }}</span>
```
Comment (kvas-damian) [inline lib/js/components/TabItem/TabItem.vue:13]: In figma, I just found that `label` is optional. We sould add `v-if` here and make prop optional

File: lib/js/components/TabItem/TabItem.vue [hunk 6]
```diff
@@ -1,67 +1,125 @@
...
+		:title="label"
+		:class="{
+			'-sizeMedium': size === TAB_ITEM_SIZES.MEDIUM,
+			'-sizeSmall': size === TAB_ITEM_SIZES.SMALL,
+			'-isSelected': isSelected,
+		}"
+		@click="$emit('click')"
+	>
+		<ds-icon v-if="icon !== null" class="tabItem__icon" :icon="icon" :size="iconSize" />
+		<span v-if="label !== null" class="tabItem__label">{{ label }}</span>
```
[the remark this answers is not in the document]
Comment (pioziol123) [reply lib/js/components/TabItem/TabItem.vue:13]: How to relax? You mean not to allow for empty string?

[the remark this answers is not in the document]
Comment (kvas-damian) [reply lib/js/components/TabItem/TabItem.vue:13]: `v-if="label"`

**Review by kvas-damian — APPROVED** [review]

**Pull request merged**